### PR TITLE
add userId to endpoint for homepage #119

### PR DESCRIPTION
**Pull request by tasneembhiri**

### CODE REVIEW

File: client/src/components/Common/Header/Header.js [hunk 1]
```diff
@@ -1,19 +1,45 @@
-import React from 'react'
+import React, { Component } from 'react'
 import GSGlog from '../../../Images/gsg_logo.jpg'
 import Profile from '../../../Images/Ellipse 2.jpg'
 import './Header.css'
 import { Link } from 'react-router-dom'
+import { withRouter } from 'react-router'
```
Comment (yaqootturman) [inline client/src/components/Common/Header/Header.js:8]: can you explain? or document this!

**Review by yaqootturman — CHANGES_REQUESTED** [review]

**Review by yaqootturman — APPROVED** [review]

**Review by Hanan795 — APPROVED** [review]

File: client/src/components/Common/Header/Header.js [hunk 2]
```diff
@@ -1,19 +1,48 @@
...
+  constructor (props) {
+    super(props)
+    this.state = {
+      email: '',
+      userId: '',
+      userName: ''
+    }
+  }
+
+  show = ()=> {
```
Comment (11janci) [inline client/src/components/Common/Header/Header.js:21]: not a very good name as it doesn't say what it's supposed to show, and actually does not show anything - `redirectToHome` or sth similar would be better.

File: client/src/components/Layout/Login/Login.js [hunk 3]
```diff
@@ -13,10 +13,11 @@ class Login extends Component {
   componentDidMount() {
     const { history } = this.props;
     axios.get("/api/check").then(({ data }) => {
-      const { success} = data;
-
-      if (success) return history.push("/home");
-    
+      const { success,userId} = data;
+      if (success)  {
+        this.setState({userId: userId})
```
Comment (11janci) [inline client/src/components/Layout/Login/Login.js:18]: do you really need to put this into the state? Seems like it is not used anywhere (unless I'm missing sth)

File: client/src/components/Common/Header/Header.js [hunk 4]
```diff
@@ -1,19 +1,48 @@
...
+
+  render () {
+    const { userId } = this.state 
+    return (
+      <div className='header'>
+        <Link to={'/'} className='header_link'>
+          <img className='logo' src={GSGlog} alt='logo' />
+        </Link>
+        <Link to = {{pathname:`/home/${userId}`}}>
+          <img className='profile' src={Profile} alt='profile' onClick={this.show}/></Link></div>
```
Comment (11janci) [inline client/src/components/Common/Header/Header.js:43]: seems there is some redundancy here. You have a link to the user's home page, but then you have the onClick where you do the redirect manually. If the `onClick` is being used, you should be able to remove the Link and just keep the img.

**Review by 11janci — COMMENTED** [review]
Pls see my comments [here](https://github.com/GSG-K3/rooms-booker/pull/93#pullrequestreview-411648157) about the check API. Would be good if you guys managed to refactor that (of course not in this PR)

**Pull request merged**